src/backend/distributed/commands/table.c

```diff
@@ -898,7 +947,40 @@ PreprocessAlterTableAddIndexConstraint(AlterTableStmt *alterTableStatement, Oid
 	ObjectAddressSet(ddlJob->targetObjectAddress, RelationRelationId, relationId);
 	ddlJob->startNewTransaction = false;
 	ddlJob->metadataSyncCommand = ddlCommand;
-	ddlJob->taskList = DDLTaskList(relationId, ddlCommand);
+
+
+	if (constraint->contype == CONSTR_FOREIGN)
+	{
+		Oid rightRelationId = RangeVarGetRelid(constraint->pktable, NoLock,
+											   false);
+
+		if (IsCitusTableType(rightRelationId, REFERENCE_TABLE))
+		{
+			SetLocalMultiShardModifyModeToSequential();
```

Comment:
I wonder, Can we easily combine this with the previous `SwitchToSequentialAndLocalExecutionIfConstraintNameTooLong`

Reply:
SetLocalMultiShardModifyModeToSequential seems specific to foreign keys/constraints involving multiple relations. I think it is better at this granularity for now.